Snake.h

```diff
@@ -0,0 +1,66 @@
+#ifndef SNAKE_H
+#define SNAKE_H
+
+#include <cassert>
+
+class Snake
+{
+public:
+    enum Direction
+    {
+        up,
+        down,
+        left,
+        right,
+        max_directions
+    };
+
+    explicit Snake(int x = 3, int y = 2, int length = 1, Direction dir = up)
+    : m_posX{x}, m_posY{y}, m_length{length}, m_dir{dir}
```

Comment:
consider to use user defined class such as coordinates instead of two separate variables x and y
try to look at std::pair or define your own class for that